OpenRA.Game/Network/Connection.cs

```diff
@@ -208,33 +221,24 @@ protected override void Send(byte[] packet)
 				}
 
 				queuedSyncPackets.Clear();
-				ms.WriteTo(socket.GetStream());
+				ms.WriteTo(tcp.GetStream());
 			}
 			catch (SocketException) { /* drop this on the floor; we'll pick up the disconnect from the reader thread */ }
 			catch (ObjectDisposedException) { /* ditto */ }
 			catch (InvalidOperationException) { /* ditto */ }
 			catch (IOException) { /* ditto */ }
 		}
 
-		bool disposed = false;
-
 		protected override void Dispose(bool disposing)
 		{
 			if (disposed)
 				return;
 			disposed = true;
 
-			t.Abort();
-			if (disposing)
-				if (socket != null)
-					socket.Client.Close();
+			if (tcp != null)
```

Comment:
Can you add a comment explaining that this will make the reader thread throw an exception, clean up after itself, and exit?